pkg/cli/completion.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+func (c *cli) newCompletionCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:                   "completion [bash|zsh|powershell]", // TODO (requires spf13.cobra update): add fish
+		DisableFlagsInUseLine: true,
+		ValidArgs:             []string{"bash", "zsh", "powershell"},
+		Args:                  cobra.ExactValidArgs(1),
+		Short:                 "Generate completion script",
+		Long: fmt.Sprintf(`To load completions:
+
+Bash:
+
+$ source <(%[1]s completion bash)
+
+# To load completions for each session, execute once:
+Linux:
+  $ %[1]s completion bash > /etc/bash_completion.d/%[1]s
+MacOS:
+  $ %[1]s completion bash > /usr/local/etc/bash_completion.d/%[1]s
```

Comment:
It should be in the usage/examples instead of it.

Reply:
Moved to each of the usage examples